File: src/main/java/org/example/kalkulationsprogramm/dto/Freigabe/FreigabeAkzeptierenRequest.java
package org.example.kalkulationsprogramm.dto.Freigabe;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

/**
 * Request-Body für POST /api/internal/freigabe/{uuid}/akzeptieren.
 *
 * <p>Vor- und Nachname sind seit der Beweissicherungs-Erweiterung Pflicht – bei
 * Firmenkunden ist das die vertretungsberechtigte Person. Die Bean-Validation
 * sorgt für HTTP 400 mit Feldname (siehe {@code RestExceptionHandler}), wenn
 * Astro nicht beide Felder mitschickt.</p>
 */
@Getter
@Setter
public class FreigabeAkzeptierenRequest
{
    /** Vom Kunden bestätigte E-Mail (optional, kommt aus dem Akzeptanz-Formular). */
    private String email;

    /** Vorname der freigebenden Person bzw. der vertretungsberechtigten Person. */
    @NotBlank(message = "Vorname ist erforderlich.")
    @Size(min = 2, max = 80, message = "Vorname muss zwischen 2 und 80 Zeichen lang sein.")
    private String vorname;

    /** Nachname der freigebenden Person bzw. der vertretungsberechtigten Person. */
    @NotBlank(message = "Nachname ist erforderlich.")
    @Size(min = 2, max = 80, message = "Nachname muss zwischen 2 und 80 Zeichen lang sein.")
    private String nachname;

    /**
     * Zusammengesetzter Anzeigename "Vorname Nachname" – wird von Astro vorberechnet
     * mitgesendet, damit serverseitig kein Format-Mismatch entsteht.
     */
    @NotBlank(message = "Unterzeichnername ist erforderlich.")
    @Size(min = 2, max = 160, message = "Unterzeichnername muss zwischen 2 und 160 Zeichen lang sein.")
    private String unterzeichnerName;

    /** Pflicht-Bestätigung "Ich habe das Dokument geprüft und nehme es verbindlich an". */
    private boolean bestaetigung;
    /** Vom Astro-Layer aus Request-Headern extrahierte echte Client-IP. */
    private String clientIp;
    /** Vom Astro-Layer extrahierter User-Agent. */
    private String userAgent;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/Freigabe/FreigabeAkzeptierenRequest.kt
package org.example.kalkulationsprogramm.dto.Freigabe

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class FreigabeAkzeptierenRequest(
    var email: String? = null,
    @field:NotBlank(message = "Vorname ist erforderlich.")
    @field:Size(min = 2, max = 80, message = "Vorname muss zwischen 2 und 80 Zeichen lang sein.")
    var vorname: String? = null,
    @field:NotBlank(message = "Nachname ist erforderlich.")
    @field:Size(min = 2, max = 80, message = "Nachname muss zwischen 2 und 80 Zeichen lang sein.")
    var nachname: String? = null,
    @field:NotBlank(message = "Unterzeichnername ist erforderlich.")
    @field:Size(min = 2, max = 160, message = "Unterzeichnername muss zwischen 2 und 160 Zeichen lang sein.")
    var unterzeichnerName: String? = null,
    var isBestaetigung: Boolean = false,
    var clientIp: String? = null,
    var userAgent: String? = null,
    var ausgewaehlteAlternativen: List<String>? = null,
)
